### DISCUSSION TITLE
How to remove a failing entity from the current entity manager?

### DISCUSSION THREAD

**Discussion by ghempton** [mikro-orm/mikro-orm#5205] [Q&A]
Consider the following code:

```
let message: Message;
try {
  message = new Message({
     user: ref(User, userId),
  });
  // This will fail because userId is a bad foreign key
  await em.persistAndFlush(message);
} catch(e) {
   // The below doesn't work as future calls to `em.flush()` will include the invalid entity
  em.remove(message);
}

// This still tries to save the bad entity above
await em.flush();
```

How would I catch errors and remove the failing entity from the entity manager? I've tried the above as well as things like `em.getUnitOfWork().unsetIdentity`. Is there a way to catch errors more gracefully than this? Using v6.

**Comment by B4nan** [comment]
You could try `em.getUnitOfWork().unsetIdentity()`. But I'd say this is the wrong way of thinking, you need to start from scratch with a fresh fork, not fix the broken one. `em.remove` is about removing an entity from the database, not removing "broken entity from context", and although it does work in the way that `em.remove` cancels the `em.persist` call, you can't do that after you flush, your entity is no longer is the persist stack (which would be canceled this way), the problem is rather about it being reachable from some other entity (it's part of the entity graph).
